test/test_torch.py

```diff
@@ -12737,6 +12807,63 @@ def test_function_unwrap_message(self):
         self.assertRaisesRegex(RuntimeError, ' call to _th_lt',
                                lambda: torch.ones(1, dtype=torch.float) < torch.ones(1, dtype=torch.double))
 
+    @torchtest.for_all_device_types()
+    def test_unary_op_mem_overlap(self, device):
+        sz = 3
+        doubles = torch.randn(2 * sz, device=device)
+        positives = torch.randint(1, 100, (2 * sz,), device=device).double()
+        ints = torch.randint(-100, 100, (2 * sz,), device=device)
+        ops = [
+            (ints, torch.bitwise_not, torch.Tensor.bitwise_not_),
+            (doubles, torch.neg, torch.Tensor.neg_),
+            (doubles, torch.abs, torch.Tensor.abs_),
+            (doubles, torch.acos, torch.Tensor.acos_),
+            (doubles, torch.asin, torch.Tensor.asin_),
+            (doubles, torch.atan, torch.Tensor.atan_),
+            (doubles, torch.ceil, torch.Tensor.ceil_),
+            (doubles, torch.cos, torch.Tensor.cos_),
+            (doubles, torch.erf, torch.Tensor.erf_),
+            (doubles, torch.erfc, torch.Tensor.erfc_),
+            (doubles, torch.exp, torch.Tensor.exp_),
+            (doubles, torch.expm1, torch.Tensor.expm1_),
+            (doubles, torch.floor, torch.Tensor.floor_),
+            (doubles, torch.frac, torch.Tensor.frac_),
+            (positives, torch.log, torch.Tensor.log_),
+            (positives, torch.log10, torch.Tensor.log10_),
+            (positives, torch.log1p, torch.Tensor.log1p_),
+            (positives, torch.log2, torch.Tensor.log2_),
+            (doubles, torch.reciprocal, torch.Tensor.reciprocal_),
+            (doubles, torch.round, torch.Tensor.round_),
+            (positives, torch.rsqrt, torch.Tensor.rsqrt_),
+            (doubles, torch.sin, torch.Tensor.sin_),
+            (doubles, torch.sigmoid, torch.Tensor.sigmoid_),
+            (doubles, torch.sqrt, torch.Tensor.sqrt_),
+            (doubles, torch.tan, torch.Tensor.tan_),
+            (doubles, torch.tanh, torch.Tensor.tanh_),
+            (doubles, torch.trunc, torch.Tensor.trunc_)
+        ]
+        for (data, op, inplace_op) in ops:
+            # internal overlap check for inplace_op
+            self.unary_check_mem_overlap(self, inplace_op, device=device)
```

Comment:
We should rename this to `unary_check_internal_mem_overlap` so it's not ambiguous.

Reply:
addressed